radis/api/geisaapi.py

```diff
@@ -321,6 +326,61 @@ def get_last(b):
     return b[non_zero]
 
 
+def add_geisa_local_quanta(df, mol=None):
+    r"""Extract branch and lower-state rotational quantum number (jl) from GEISA local quanta.
+
+    This helper is meant for GEISA diatomic / linear molecules (e.g. CO), where the
+    local lower-state quantum string ``locl`` encodes the branch and J in a compact
+    form such as::
+
+        "R  0", "R 12", "P 59", "P130", ...
+
+    The function parses:
+
+    - ``branch`` as the first non-space character (``"P"``, ``"Q"``, ``"R"``, ...),
+    - ``jl`` as the integer formed by the remaining characters.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        DataFrame returned by :func:`gei2df`, containing at least a ``locl`` column.
+    mol : str, optional
+        Molecule name. If not in HITRAN_GROUP2 (diatomic / linear), the DataFrame
+        is returned unchanged.
+
+    Returns
+    -------
+    pandas.DataFrame
+        The same DataFrame with two additional columns:
+
+        - ``branch`` : str
+        - ``jl`` : int
+
+    Notes
+    -----
+    This is a GEISA-specific post-processing step. It mirrors, in a simplified way,
+    the HITRAN/HITEMP treatment of local quanta for diatomic / linear molecules,
+    but uses the GEISA ``locl`` layout instead of HITRAN fixed-width slices.
+    """
+    # Only apply to diatomic / linear molecules (HITRAN Group 2)
+    if mol is None or mol.upper() not in HITRAN_GROUP2:
```

Comment:
```suggestion
    if mol is None or mol not in HITRAN_GROUP2:
```
`.upper()` will cause inconsistency for eg HCl will become HCL.